### PR TITLE
feat:[CDS-88165]: Add logs to check how many secrets does not have pe…

### PR DESCRIPTION
**Pull request by Aninditaa-ch**
…rmission to access secrets using SecretFunctor

## Describe your changes
Changes added to get the logs for the case in which user without the secrets permission is running the pipeline.
Testing: Tested that the logs were being registered with the necessary information. Changes were implemented as required, after running the pipeline. 

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/57394)

### CODE REVIEW

**Comment by sahilhindwani-harness** [comment]
Please add the following to PR description:
- What testing was done? If no testing was done, please mention the same on the ticket.

File: pipeline-service/modules/orchestration/src/main/java/io/harness/engine/expressions/functors/SecretFunctor.java [hunk 1]
```diff
@@ -9,36 +9,69 @@
...
     this.secretsRuntimeUsagesSubject = secretsRuntimeUsagesSubject;
   }
 
   public Object getValue(String secretIdentifier) {
-    if (EmptyPredicate.isNotEmpty(secretIdentifier) && ambiance != null
-        && AmbianceUtils.shouldEnableSecretsObserver(ambiance)) {
-      secretsRuntimeUsagesSubject.fireInform(SecretResolutionObserver::onSecretsRuntimeUsage,
-          SecretObserverInfo.builder().secretIdentifier(secretIdentifier).ambiance(ambiance).build());
-    }
+    try (PmsSecurityContextEventGuard securityContextEventGuard = new PmsSecurityContextEventGuard(ambiance)) {
```
Comment (sahilhindwani-harness) [inline pipeline-service/modules/orchestration/src/main/java/io/harness/engine/expressions/functors/SecretFunctor.java:52]: This seems to be not in sync with the changes mentioned in the ticket, could you please check once?

**Review by sahilhindwani-harness — CHANGES_REQUESTED** [review]

**Comment by Aninditaa-ch** [comment]
trigger uts

**Review by Aninditaa-ch — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57394)* status: 1 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/engine/expressions/functors/SecretFunctor.java` line 52 at r1](https://reviewable.io/reviews/harness/harness-core/57394#gh-1446967160:-Nnqz3m7CHDC2qUILjIu:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2a7c076606ca12294cd7ee863edada077871842a/pipeline-service/modules/orchestration/src/main/java/io/harness/engine/expressions/functors/SecretFunctor.java#L52)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

This seems to be not in sync with the changes mentioned in the ticket, could you please check once?
</blockquote></details>

Done.

**Review by sahilhindwani-harness — CHANGES_REQUESTED** [review]

**Comment by Aninditaa-ch** [comment]
trigger SmartPRChecks
trigger uts

**Comment by Aninditaa-ch** [comment]
run code-formatter

**Comment by Aninditaa-ch** [comment]
trigger code-formatter
trigger uts

**Comment by Aninditaa-ch** [comment]
trigger uts

**Review by sahilhindwani-harness — DISMISSED** [review]

**Comment by Aninditaa-ch** [comment]
trigger ut8

**Comment by Aninditaa-ch** [comment]
trigger uts

**Comment by Aninditaa-ch** [comment]
trigger uts

**Review by sahilhindwani-harness — APPROVED** [review]

**Review by ankittiwari-harness — APPROVED** [review]

**Pull request merged**